akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala

```diff
@@ -57,69 +64,202 @@ private[http] object Handshake {
             list of values indicating which extensions the client would like
             to speak.  The interpretation of this header field is discussed
             in Section 9.1.
-   */
-  def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
-    def find[T <: HttpHeader: ClassTag]: Option[T] =
-      headers.collectFirst {
-        case t: T ⇒ t
+    */
+    def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
+      def find[T <: HttpHeader: ClassTag]: Option[T] =
+        headers.collectFirst {
+          case t: T ⇒ t
+        }
+
+      // val host = find[Host]
```

Comment:
Yes please, a small comment on these is enough ("because can be handled by outer directive" or something like that) I "got" that one rather quickly, but maybe someone else would have been puzzled.